### PR TITLE
[BALDE-3] File Manager

### PR DESCRIPTION
**Pull request by Blinkuu**

### CODE REVIEW

File: src/utils/file_manager.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+* @class FileManager
+* @brief implementation of file manager system, if you are on Windows add -D_WINDOWS definition, if on Linux add -D_LINUX definition
+*/
+
+    class FileManager {
+    public:
+        enum Size : char {
+            SMALL, BIG
```
Comment (piotrmoszkowicz) [inline src/utils/file_manager.h]: If you ask me I would prefer to have better naming like BIG_FILE, SMALL_FILE

File: src/utils/file_manager.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        /**
+         * @fn ReadFile
+         * @brief use to read files
+         * @return string
+         */
+
+        [[nodiscard]] static std::string ReadFile(const char* filePath, Size size);
+    private:
+        [[nodiscard]] static std::string ReadSmallFile(const char* filePath); /** used to read small files, same implementation on windows and linux*/
```
Comment (piotrmoszkowicz) [inline src/utils/file_manager.h]: Why different documentation style ?

File: src/utils/file_manager.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        [[nodiscard]] static std::string ReadBigFile(const char* filePath); /** used to read big files, different implementation for windows and linux*/
```
Comment (piotrmoszkowicz) [inline src/utils/file_manager.h]: Why different documentation style ?

File: src/utils/file_manager.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        if (!file) {
+            CORE_LOG_WARN("Couldn't open the file at path: " + static_cast<std::string>(filePath));
+            return std::string("Error!");
+        }
+
+        fseek(file, 0, SEEK_END);
+        auto stringSize = static_cast<unsigned long>(ftell(file));
+        fseek(file, 0, SEEK_SET);
+
+        auto *buffer = new char[stringSize + 1];
```
Comment (piotrmoszkowicz) [inline src/utils/file_manager.cpp]: What happens when we do not have memory and buffer == null? Add check for that.

**Review by piotrmoszkowicz — CHANGES_REQUESTED** [review]
Mostly nice, but some changes are needed.

File: src/utils/file_manager.cpp (hunk 4 above)
Comment (Blinkuu) [reply src/utils/file_manager.cpp]: operator new throws a std::bad_alloc exception on failure, unless you're explicitly using the nothrow version

**Review by piotrmoszkowicz — APPROVED** [review]

**Review by BIGbadEL — APPROVED** [review]

**Pull request merged**